gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FlowSpecSearchObject.java

```diff
@@ -191,5 +202,12 @@ public void completePreparedStatement(PreparedStatement statement)
     if (this.getOwningGroup() != null) {
       statement.setString(++i, this.getOwningGroup());
     }
+
+    if (this.getCount() != 0) {
```

Comment:
If you define the null value here to be -1 which I see above, your boolean check should represent that value.